### PR TITLE
refactor: :art: appwrite client code is refactored

### PR DESCRIPTION
**Pull request by gopal1996**

### CODE REVIEW

File: src/server/data-access/session.ts [hunk 1]
```diff
@@ -3,27 +3,24 @@ import "server-only";
...
-  createAdminClient,
-  createSessionClient,
   getUniqueID,
   oAuthProvidersType,
 } from "@/server/services/appwrite";
+import { serviceClient } from "../services";
 
 export async function getSession() {
-  const { account } = await createSessionClient();
+  const account = (await serviceClient.createSession()).account;
```
Comment (sadanandpai) [inline src/server/data-access/session.ts]: `const { account } = await serviceClient.createSession();`

is this not supported

File: src/server/data-access/submissions.ts [hunk 2]
```diff
@@ -18,7 +19,7 @@ export async function createSubmissionsRecord(
   code: string,
   status: boolean
 ) {
-  const { client } = await createSessionClient();
+  const client = (await serviceClient.createSession()).client;
```
Comment (sadanandpai) [inline src/server/data-access/submissions.ts]: as we are working on the abstraction, can we also abstract out the database object part as well

`const { databases } = await serviceClient.createSession()`

File: src/server/services/appwrite.ts [hunk 3]
```diff
@@ -15,80 +15,78 @@ export function getUniqueID() {
   return ID.unique();
 }
 
-export function getOAuthProvider(provider: keyof typeof oAuthProviders) {
-  return oAuthProviders[provider];
-}
+export class AppWrite {
+  clientInstance;
```
Comment (sadanandpai) [inline src/server/services/appwrite.ts]: is there any risk
https://appwrite.io/docs/tutorials/nextjs-ssr-auth/step-3

image

File: src/server/data-access/session.ts (hunk 1 above)
Comment (gopal1996) [reply src/server/data-access/session.ts]: no. this will work as well

File: src/server/data-access/submissions.ts (hunk 2 above)
Comment (gopal1996) [reply src/server/data-access/submissions.ts]: sure

File: src/server/services/appwrite.ts (hunk 3 above)
Comment (gopal1996) [reply src/server/services/appwrite.ts]: will check this

**Review by sadanandpai — APPROVED** [review]

**Review by sadanandpai — APPROVED** [review]

**Pull request merged**